examples/benchmarks/bubblebench/simple-tree/src/appState.ts

```diff
@@ -0,0 +1,76 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import {
+	type IAppState,
+	type IBubble,
+	type IClient,
+	makeBubble,
+	randomColor,
+} from "@fluid-example/bubblebench-common";
+import type { InsertableTypedNode, TreeView } from "@fluidframework/tree/internal";
+import { type App, type Bubble, Client } from "./schema.js";
+
+export class AppState implements IAppState {
+	readonly localClient: Client;
+
+	constructor(
+		private readonly tree: TreeView<App>,
+		public width: number,
+		public height: number,
+		numBubbles: number,
+	) {
+		this.tree.root.clients.insertAtEnd(
+			this.createInitialClientNode(numBubbles) as unknown as Client,
+		);
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const lastClient = this.tree.root.clients.at(this.tree.root.clients.length - 1)!;
+		this.localClient = lastClient;
+
+		console.log(
+			`created client with id ${this.localClient.clientId} and color ${this.localClient.color}`,
+		);
+	}
+
+	public applyEdits() {}
+
+	createInitialClientNode(numBubbles: number): IClient {
+		const bubbles: IBubble[] = [];
+		// create and add initial bubbles to initial client json tree
+		for (let i = 0; i < numBubbles; i++) {
+			const bubble = makeBubble(this.width, this.height);
+			bubbles.push(bubble);
+		}
+
+		const client: IClient = {
+			clientId: `${Math.random()}`,
+			color: randomColor(),
+			bubbles,
+		};
+
+		return client;
+	}
+
+	public get clients() {
+		return [...this.tree.root.clients];
```

Comment:
This doesn't seem right. If the user of this needs to mutate the array of clients, returning a copy of it here will make that not work. If they don't need to mutate it, then it should be a readonly array and this should be able to return this.tree.root.clients without copying.

Sadly IAppState doesn't document how its used, but the type it uses explicitly has some mutation methods, so it would be surprising if it didn't mutate the returned array expecting it to do something.

Reply:
I forcefully casted `this.tree.root.clients` to `IArrayish`. This seems to be a safe method to incorporate certain mutation methods.

Reply:
I don't think our array nodes implement push, pop or map in a compatible way so this is not safe. If they did this would not require an explicit cast.

I did a little digging into this code and it seems users of IAppState.clients don't actually need to mutate it. So I think the fix here is to simply remove mutability from the type required here.